src/main/java/org/terasology/web/client/ClientSecurityInfo.java

```diff
@@ -15,21 +15,31 @@
  */
 package org.terasology.web.client;
 
+import org.terasology.web.serverAdminManagement.PermissionType;
+
+import java.util.List;
+
 public class ClientSecurityInfo {
 
     private boolean isAuthenticated;
-    private boolean hasAdminPermission;
+    private boolean isAdmin;
+    private List<PermissionType> clientPermissions;
```

Comment:
I think a `Set` would be more suited since the order is not relevant, correctly? If you use a set and also replace `Map<Something, Boolean>` with `Set<Something>` as suggested in the other comments (if it's possible) you may be able to do various simplifications of the code since you won't need to do conversions between collection types (by iterating over the values) anymore

(my fault since I suggested you `List`, but I was just thinking of a generic collection, sorry)